scene/animation/animation_node_state_machine.cpp

```diff
@@ -901,18 +901,20 @@ bool AnimationNodeStateMachinePlayback::_transition_to_next_recursive(AnimationT
 	bool is_state_changed = false;
 
 	NextInfo next;
-	StringName transition_start = current;
+	Vector<StringName> transition_path;
```

Comment:
What's the usual number of elements in this vector? It looks like a use case for the hash map, but I guess it's quite small and won't benefit from `HashSet<StringName>` usage?

Reply:
Depending on the complexity of the StateMachine, but I guess the number of "transition path in one frame" should not exceed 10 at the most.